### PR TITLE
Fixing media browser issues (#15129)

### PR DESCRIPTION
**Pull request by wax100**
### What does it do?
fix downloding file name (earlier was index)
fix upload file window (destroy and create new instead of creating, creating, creating...)
fix deleting more then 1 file

### Related issue(s)/PR(s)
issue #15129

### CODE REVIEW

File: core/src/Revolution/Processors/Browser/File/Download.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function process()
         @session_write_close();
         try {
             if ($data = $this->source->getObjectContents($file)) {
-                $name = preg_replace('#[^\w-.]#ui', '_', $data['basename']);
+                $name = $data['basename'];//preg_replace('#[^\w-.]#ui', '_', $data['basename']);
```
Comment (Ibochkarev) [inline core/src/Revolution/Processors/Browser/File/Download.php]:
```suggestion
                $name = $data['basename'];
```

**Review by Ibochkarev — CHANGES_REQUESTED** [review]

**Comment by Mark-H** [comment]
Can you please try to use descriptive descriptions in commit messages and PRs? What it changes is much more important than the related issue number ;)

**Review by Ibochkarev — APPROVED** [review]
https://user-images.githubusercontent.com/2138260/110111547-b3ee1800-7dda-11eb-99dd-bcc979765888.mp4

**Comment by Ruslan-Aleev** [comment]
If you delete a file and immediately load another (or create a file / category), the file manager breaks down.

file_del

There are errors in the browser console:

```
Uncaught TypeError: Cannot read property 'getSelectionModel' of null
    at Ext.tree.AsyncTreeNode.removeChild (ext-all.js:21)
    at MODx.tree.TreeLoader.load (ext-all.js:21)
    at MODx.tree.Directory.refreshActiveNode (modx.tree.js:712)
    at MODx.tree.Directory.fn (modx.tree.directory.js:589)
    at y.Event.fire (ext-all.js:21)
    at MODx.window.CreateDirectory.fireEvent (ext-all.js:21)
    at MODx.window.CreateDirectory.success (modx.window.js:249)
    at Object.callback (ext-all.js:21)
    at constructor.afterAction (ext-all.js:21)
    at Ext.form.Action.Submit.success (ext-all.js:21)
```

**Comment by Ibochkarev** [comment]
> If you delete a file and immediately load another (or create a file / category), the file manager breaks down.
> 
> file_del
> 
> There are errors in the browser console:
> 
> ```
> Uncaught TypeError: Cannot read property 'getSelectionModel' of null
>     at Ext.tree.AsyncTreeNode.removeChild (ext-all.js:21)
>     at MODx.tree.TreeLoader.load (ext-all.js:21)
>     at MODx.tree.Directory.refreshActiveNode (modx.tree.js:712)
>     at MODx.tree.Directory.fn (modx.tree.directory.js:589)
>     at y.Event.fire (ext-all.js:21)
>     at MODx.window.CreateDirectory.fireEvent (ext-all.js:21)
>     at MODx.window.CreateDirectory.success (modx.window.js:249)
>     at Object.callback (ext-all.js:21)
>     at constructor.afterAction (ext-all.js:21)
>     at Ext.form.Action.Submit.success (ext-all.js:21)
> ```

I confirm. Didn't look

**Review by Ibochkarev — CHANGES_REQUESTED** [review]
You must remove the commit Merge branch 'bug-15129' of https://github.com/wax100/revolution into ...

**Comment by Ruslan-Aleev** [comment]
@wax100 The error with uploading a file after deleting is no longer reproducible (https://github.com/modxcms/revolution/pull/15460#issuecomment-791373650), but persists when trying to create a category after deleting a file:

media_bug_cat

**Comment by theboxer** [comment]
@wax100 any plans on resolving the issues reported here so we can merge?

**Pull request closed without merging**